### PR TITLE
storage usage audit: set duration and timestamp of last run as metrics

### PR DESCRIPTION
**Pull request by pH14**
Inspired by discussion in: https://materializeinc.slack.com/archives/CL68GT3AT/p1684866034791529

Creates two metrics: gauges to record the duration and epoch timestamp of the last run of the storage usage audit. These will be helpful to understand at-a-glance the frequency of the audit, and potentially to alert if it ever takes a concerningly long time to execute / we go a long time without a new datapoint.

I chose to remove the existing histogram `storage_usage_collection_time_seconds` because: 1) I couldn't find it used anywhere in grafana 2) its bucket granularity only measured runs between 1.28ms and 8s... but the audit in practice can be measured in minutes to hours 3) a gauge is much higher precision / generally seems like a better fit for low frequency events like the audit.

cc @necaris @arusahni 

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: src/adapter/src/metrics.rs [hunk 1]
```diff
@@ -59,10 +60,13 @@ impl Metrics {
                 help: "The total number of adapter commands issued of the given type since process start.",
                 var_labels: ["command_type", "status", "application_name"],
             )),
-            storage_usage_collection_time_seconds: registry.register(metric!(
-                name: "mz_storage_usage_collection_time_seconds",
-                help: "The number of seconds the coord spends collecting usage metrics from storage.",
-                buckets: histogram_seconds_buckets(0.000_128, 8.0)
+            storage_usage_collection_last_run_duration: registry.register(metric!(
```
Comment (mjibson) [inline src/adapter/src/metrics.rs:63]:
```suggestion
            storage_usage_collection_last_run_seconds: registry.register(metric!(
```

Comment (mjibson) [reply src/adapter/src/metrics.rs:63]: Recommend changing this to seconds, see https://prometheus.io/docs/practices/naming/.

**Review by mjibson — APPROVED** [review]

File: src/adapter/src/metrics.rs (hunk 1 above)
Comment (mjibson) [reply src/adapter/src/metrics.rs:63]: (Or maybe `_duration_seconds`)

**Comment by pH14** [comment]
Tabling this -- we rewrote how storage usage audits work and it's many many times faster than before, so these metrics are less relevant.

**Pull request closed without merging**